src/librustc_mir/borrow_check/mutability_errors.rs

```diff
@@ -257,10 +259,11 @@ impl<'a, 'gcx, 'tcx> MirBorrowckCtxt<'a, 'gcx, 'tcx> {
                         _,
                     ) = pat.node
                     {
-                        err.span_suggestion(
+                        err.span_suggestion_with_applicability(
                             upvar_ident.span,
                             "consider changing this to be mutable",
                             format!("mut {}", upvar_ident.name),
+                            Applicability::MachineApplicable,
```

Comment:
~~"Upvar" implies this is only for closures and closure-like things (maybe generators?), right? I'm asking because I'm not sure I'd want use to change function signatures -- which are potentially public API. Actually, scratch that, if it's a valid fix, we should fix it.~~